src/actions/Entities/VaccinePrescriptionActions.js

```diff
@@ -47,25 +57,70 @@ const saveEditing = prescription => ({
   payload: { prescription },
 });
 
+const selectVaccine = vaccine => ({
+  type: VACCINE_PRESCRIPTION_ACTIONS.SELECT_VACCINE,
+  payload: { vaccine },
+});
+const selectBatch = itemBatch => ({
+  type: VACCINE_PRESCRIPTION_ACTIONS.SELECT_BATCH,
+  payload: { itemBatch },
+});
+
 const updateEditing = (value, field) => (dispatch, getState) => {
   const newVaccinePrescriptionId = selectEditingVaccinePrescriptionId(getState());
   dispatch(update(newVaccinePrescriptionId, field, value));
 };
 
+const confirm = () => (dispatch, getState) => {
+  const { user } = getState();
+  const { currentUser } = user;
+  const patientID = selectEditingNameId(getState());
+  const patient = UIDatabase.get('Name', patientID);
+  const selectedBatches = selectSelectedBatches(getState());
+
+  UIDatabase.write(() => {
+    const prescription = createRecord(
+      UIDatabase,
+      'CustomerInvoice',
+      patient,
+      currentUser,
+      'dispensary'
+    );
+
+    selectedBatches.forEach(itemBatch => {
+      const { item } = itemBatch;
+      const transactionItem = createRecord(UIDatabase, 'TransactionItem', prescription, item);
```

Comment:
A transaction *item* is really just a grouping of transaction *batches*, which, when setting the total quantity will create a new transaction batch using item batches in fefo order- so we need to create a transaction batch for this transaction item from the item batch the user selects 

Did it here with some testing - hope it's right 🤣 

a3fb275